### PR TITLE
Add cashflow queries

### PR DESCRIPTION
**Pull request by sanghviharshit**
(Also includes some auto-formatting changes)

### CODE REVIEW

File: monarchmoney/monarchmoney.py [hunk 1]
```diff
@@ -418,19 +419,175 @@ async def get_transaction_tags(self) -> Dict[str, Any]:
...
+      variables["filters"]["endDate"] = datetime(datetime.now().year, datetime.now().month, last_date).strftime("%Y-%m-%d")
+
+    return await self.gql_call(
+        operation="Web_GetCashFlowPage",
+        variables=variables,
+        graphql_query=query
+    )
+
+  async def get_cashflow_summary(self,
+      start_date: Optional[str]=None,
```
Comment (hammem) [inline monarchmoney/monarchmoney.py:529]: for consistency

```suggestion
  async def get_cashflow_summary(
      self,
      limit: int=1000,
      start_date: Optional[str]=None,
```

File: monarchmoney/monarchmoney.py [hunk 2]
```diff
@@ -418,19 +419,175 @@ async def get_transaction_tags(self) -> Dict[str, Any]:
...
+            savingsRate
+            __typename
+          }
+          __typename
+        }
+      }
+    """)
+
+    variables = {
+      "limit": 25,
```
Comment (hammem) [inline monarchmoney/monarchmoney.py:551]:
```suggestion
      "limit": limit,
```

File: monarchmoney/monarchmoney.py [hunk 3]
```diff
@@ -418,19 +419,175 @@ async def get_transaction_tags(self) -> Dict[str, Any]:
...
+      variables["filters"]["startDate"] = start_date
+      variables["filters"]["endDate"] = end_date
+    elif bool(start_date) != bool(end_date):
+      raise Exception("You must specify both a startDate and endDate, not just one of them.")
+    else:
+      current_year = datetime.now().year
+      current_month = datetime.now().month
+      last_date = calendar.monthrange(current_year, current_month)[1]
+      variables["filters"]["startDate"] = datetime(current_year, current_month, 1).strftime("%Y-%m-%d")
+      variables["filters"]["endDate"] = datetime(datetime.now().year, datetime.now().month, last_date).strftime("%Y-%m-%d")
```
Comment (hammem) [inline monarchmoney/monarchmoney.py:571]: same question about using a future date

```suggestion
      variables["filters"]["endDate"] = datetime(current_year,  current_month, last_date).strftime("%Y-%m-%d")
```

**Review by hammem — CHANGES_REQUESTED** [review]
Thanks for putting this together, @sanghviharshit !  Really appreciate adding the cash flow calls.

Some small stuff inline below, almost all about consistency with the existing API functions.  

Also a question about the need to specify date range params by default, as the rest of the Monarch gql calls don't require them and have default behavior.

File: monarchmoney/monarchmoney.py [hunk 4]
```diff
@@ -418,19 +419,175 @@ async def get_transaction_tags(self) -> Dict[str, Any]:
...
```
[the remark this answers is not in the document]
Comment (sanghviharshit) [reply monarchmoney/monarchmoney.py:520]: Yes, that's how the cash flow page on the website queries it for the current month.

File: monarchmoney/monarchmoney.py (hunk 3 above)
Comment (sanghviharshit) [reply monarchmoney/monarchmoney.py:575]: Yes, this is how it's queried on the website for current month

**Comment by sanghviharshit** [comment]
> Also a question about the need to specify date range params by default, as the rest of the Monarch gql calls don't require them and have default behavior.

You are right, the dates in the gql are optional, but it doesn't give predictable results. it returns the cashflow for the previous month when no dates specified. I think setting defaults to the current month keeps it predictable for the caller (e.g. Home Assistant) so they don't need to pass start_date, end_date to get latest cashflow data

**Review by hammem — APPROVED** [review]
Thanks, @sanghviharshit !

**Pull request merged**